src/docs/arch-backup.dox

```diff
@@ -0,0 +1,69 @@
+/*! @arch_page arch-backup Backup
+
+# Overview #
+
+Backup in WiredTiger can be performed using the \c "backup:" cursor. WiredTiger
+supports four types of backups.
+
+1. Full database
+2. Block-based incremental backup
+3. Log-based incremental backup
+4. Target backup
+
+The following page describes how backup works inside the WiredTiger. Please refer @ref backup
+for details on how to use each of these types of backups.
+
+# Full database #
+
+When the application opens the backup cursor, internally WiredTiger generates a list of all
+files in the database that are necessary for the backup. When generating that list WiredTiger
+takes both the checkpoint and schema locks to block database file modifications. There is a
+contract that all files that exist at the time the backup starts exist for the entire time
+the backup cursor is open. This contract applies to all files in the database, not just files
+that are included in the generated list.
+
+WiredTiger log files are also part of the generated file list for backup. There is a contract
+that the log files do not contain or make visible operations that are added to the log after
+the backup started that could have adverse affects during recovery and restore. Therefore
+WiredTiger forces a log file switch when opening the backup cursor. To fulfill the earlier
+contract that all files must exist during backup, WiredTiger does not use any pre-allocated
+log files nor does it archive or remove old log files during the time the backup cursor is open.
+
+Once the backup is opened successfully, any checkpoints that exist on the database before
+backup starts must be retained until the backup cursor is closed. All the newer checkpoints
+that are created during the backup in progress are cleaned whenever a newer checkpoint is
+created.
+
+# Block-based incremental backup #
+
+Block-based incremental backup is performed by tracking the modified blocks in the checkpoint.
+Whenever a new checkpoint occurs on a file, any new or modified blocks in this checkpoint are
+recorded as a bit string. This bit string information is updated with every new checkpoint.
+
+Whenever the incremental backup cursor is opened to perform the backup. WiredTiger returns all
+the file offsets and sizes that are modified after the full backup. WiredTiger returns the full
```

Comment:
'after the full backup' is not quite right because the modified blocks could be from the full backup or simply an earlier (i.e. the previous) incremental backup. I suggest:
WiredTiger returns all the file offsets and sizes that are modified from the previous source backup identifier.

Reply:
Fixed as suggested.